LabelsRandomColors2.0/src/com/company/Main.java

```diff
@@ -46,10 +44,12 @@ private void newLabel(String s) {
         GLabel label = new GLabel(s);
         label.setColor(rgen.nextColor());
         label.setFont("ARIAL-18");
+
         add(label, rgen.nextDouble(0, (getWidth() - label.getWidth())),

```

Comment:
Faccio un po' fatica a capire cosa sta succedendo qua. Forse potresti aumentare la leggibilita' se 'nascondessi' i dettagli della posizione random in una classe? Tipo cosi':

```java
...

RandomPosition randomPosition = new RandomPosition(label)

add(label, randomPosition.getX(), randomPosition.getY())

...
```

Reply:
O, ancora meglio, potresti create un `GPoint` e poi fare

```java
GPoint randomPoint = generateRandomPoint(label)
add(label, randomPoint)
```

Reply:
Ottimo, 
non sapevo del metodo randomPoint, Grazie

Reply:
eh pero' non esiste, lo devi creare tu 👍

Reply:
Grazie mille
prima pensavo che creare un metodo per una sola riga di codice non fosse efficiente, ma immagino cosi aiuti